### PR TITLE
Prepare engine for deprecation of async_minitest.dart

### PR DESCRIPTION
**Pull request by lrhn**
The Dart SDK's `package:async_helper/async_minitest.dart` was never intended for new tests, it was a drop-in polyfill for the legacy `package:unit_test` that some Dart SDK tests were using. The tests should never have used that package, and migrating them to `package:test` was not an option, so the `async_minitest.dart` file was the minimal polyfill that made all the tests run.

Since then, new tests have been written using the file, and to stop that, the file will now become deprecated, and (hopefully some day) removed.

If the Flutter engine wants to keep using the file, they should own their own copy of it, which is what this change does.

(Also migrates off the to-be-deprecated `assertStatementsEnabled` property of `package:expect/expect.dart`, moved to the new `asserts` from `package:expect/variations.dart`.)

### CODE REVIEW

File: testing/litetest/lib/src/async_minitest.dart [hunk 1]
```diff
@@ -0,0 +1,338 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
```
Comment (chinmaygarde) [inline testing/litetest/lib/src/async_minitest.dart:1]: This copyright header needs to be reassigned.

**Review by chinmaygarde — APPROVED** [review]
Other than the note about copyright header blocker, lgtm.

**Review by jason-simmons — APPROVED** [review]

**Comment by lrhn** [comment]
The Fuchsia builder's source-validation seems to insist that every file in a third-party package is listed in sources,
even if it's not used (and should not be used). 
That means that when the file is eventually removed, the build script will need to be updated at that time.

**Comment by chinmaygarde** [comment]
May I land this @lrhn ?

**Comment by lrhn** [comment]
Please do. 
I think I've addressed all compilation problems.

**Pull request merged**